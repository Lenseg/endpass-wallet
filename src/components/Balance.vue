<template>
	<div>
    <p class="heading">Balance</p>
    <span class="title" :class="{'long-number': balance.length > 6}">{{ balance }}</span>
    {{currency}}
    <a v-if="hasUpdate" @click.prevent="update()">
    	<span class="icon is-small"
                v-html="require('@/img/reload.svg')"></span>
    </a>
	</div>
</template>
<script>
import { BigNumber } from 'bignumber.js';
export default {
  props: {
    amount: {
      default: 0,
    },
    price: {
      default: 1,
    },
    currency: {
      type: String,
      default: 'ETH',
    },
    decimals: {
      default: 18,
    },
  },
  computed: {
    balance() {
      let amountBn;
      if (!BigNumber.isBigNumber(this.amount)) {
        amountBn = new BigNumber(this.amount);
      } else {
        amountBn = this.amount;
      }
      let priceBn;
      if (!BigNumber.isBigNumber(this.price)) {
        priceBn = new BigNumber(this.price);
      } else {
        priceBn = this.price;
      }

      let balance = amountBn.times(priceBn);
      let balanceString = balance
        .toFixed(this.decimals)
        .match(/^[0-9]{1,18}(\.[0-9]{0,18}[^0]{1,18}){0,1}/);
      balanceString = balanceString ? balanceString[0] : '0';
      return balanceString;
    },
    hasUpdate() {
      return this.$listeners.update;
    },
  },
  methods: {
    update() {
      this.$emit('update');
    },
  },
};
</script>
<style>
</style>
