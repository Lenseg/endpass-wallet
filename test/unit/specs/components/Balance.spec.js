import { shallow } from '@vue/test-utils';

import Balance from '@/components/Balance';

describe('Balance', () => {
  describe('render', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(Balance, {
        propsData: {
          amount: '12',
        },
      });
    });

    it('should be a Vue component', () => {
      expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('should render props', () => {
      const currency = 'KEK';
      const props = {
        currency,
      };

      wrapper.setProps(props);

      expect(wrapper.find('.currency').text()).toBe(currency);
    });
  });
});
