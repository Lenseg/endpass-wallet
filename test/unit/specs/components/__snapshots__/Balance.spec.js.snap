// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance render should render initial state of the component 1`] = `
<div
  class="balance has-spinner"
>
  <span
    class="title amount"
    title="12"
  >
    12
  </span>
   
  <span
    class="currency"
  >
    
      ETH
    
  </span>
   
  <!---->
   
  <!---->
</div>
`;
